crates/fj-interop/src/status_report.rs

```diff
@@ -2,25 +2,31 @@
 
 /// Struct to store and update status messages
 pub struct StatusReport {
-    status: String,
+    status: Vec<String>,
 }
 
 impl StatusReport {
     /// Create a new ``StatusReport`` instance with a blank status
     pub fn new() -> Self {
-        Self {
-            status: String::new(),
-        }
+        Self { status: Vec::new() }
     }
 
     /// Update the status
     pub fn update_status(&mut self, status: &str) {
-        self.status = status.to_string();
+        if self.status.len() >= 5 {
+            self.clear_status();
+        }
```

Comment:
I think this implementation is a bit weird, as it results in lots of statuses being removed at once. Might be better to just remove the oldest status, until only 5 are left. Maybe [`VecDeque`](https://doc.rust-lang.org/std/collections/struct.VecDeque.html) is a better type for this than `Vec`, as it provides convenient `push_back`/`pop_front` methods.

But that's something that can be fixed later.

Reply:
No problem! Thanks for the feedback! Your right, VecDeque does look convenient. I'll look into it.